File: Anul 2/Semestrul 2/MPP/labs/Tema 5-proiect Server Csharp-Client Java/proiectSCcsharp/angajatModel/validators/BiletValidator.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Xml.Linq;

namespace proiectSCcsharp.domain.validators
{
    public class BiletValidator : Validator<Bilet> 
    {
        public BiletValidator()
        {
        }

        public void validate(Bilet entity)
        {
            String errors = "";
            if (entity.cumparatorName == "")
                errors += "invalid cumparator name\n";
            if (entity.idSpectacol < 0)
                errors += "invalid id spectacol\n";
            if (entity.nrSeats < 0)
                errors += "invalid nr of seats\n";
            if (errors.Length > 0)
                throw new ValidationException(errors);
        }
    }
}

File: Anul 2/Semestrul 2/MPP/labs/Tema 7-servicii REST/Java/Rest/proiect/Domain/src/main/java/proiect/domain/validators/BiletValidator.java
package proiect.domain.validators;

import proiect.domain.Bilet;

import java.util.Objects;

public class BiletValidator implements Validator<Bilet> {
    private static final BiletValidator instance = new BiletValidator();

    public static BiletValidator getInstance() {
        return instance;
    }

    public BiletValidator() {
    }

    @Override
    public void validate(Bilet entity) throws ValidationException {
        String errors = "";
        if (Objects.equals(entity.getCumparatorName(), ""))
            errors += "invalid cumparator name\n";
        if (entity.getIdSpectacol()<0)
            errors += "invalid id spectacol\n";
        if (entity.getNrSeats()<0)
            errors += "invalid nr of seats\n";
        if (errors.length() > 0)
            throw new ValidationException(errors);
    }
}